### PR TITLE
zfs-replicate: bump version 3.1.0

### PR DESCRIPTION
**Pull request by alunduil**
## Description of changes

Version bump as requested in #250750 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/backup/zfs-replicate/default.nix [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
 
 buildPythonApplication rec {
   pname = "zfs-replicate";
-  version = "1.2.3";
+  version = "3.1.0";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "b2cb9d4670a6e12d14a446c10d857862e91af6e4526f607e08b41bde89953bb8";
+    sha256 = "24816c3e92ea9abc03e1ce281b5fcce4dfb5534092096e3e2c57a0d585ec4d67";
```
Comment (pbsds) [inline pkgs/tools/backup/zfs-replicate/default.nix:16]: Please use the new hash format, `hash = "sha256-xxxx";`

File: pkgs/tools/backup/zfs-replicate/default.nix [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
   };
 
   postPatch = ''
-    sed -i setup.cfg \
-      -e '/--cov.*/d'
+    sed -i pyproject.toml -e '/--cov[^"]*/d'
   '';
 
   nativeCheckInputs = [
```
Comment (pbsds) [inline pkgs/tools/backup/zfs-replicate/default.nix:23]:
```suggestion
  nativeCheckInputs = [
    pytestCheckHook
```

File: pkgs/tools/backup/zfs-replicate/default.nix [hunk 3]
```diff
@@ -29,7 +33,7 @@ buildPythonApplication rec {
   doCheck = true;
 
   checkPhase = ''
```
Comment (pbsds) [inline pkgs/tools/backup/zfs-replicate/default.nix:35]:
```suggestion
```

File: pkgs/tools/backup/zfs-replicate/default.nix [hunk 4]
```diff
@@ -29,7 +33,7 @@ buildPythonApplication rec {
...
-    pytest --doctest-modules
+    pytest
   '';
```
Comment (pbsds) [inline pkgs/tools/backup/zfs-replicate/default.nix:37]:
```suggestion
```

File: pkgs/tools/backup/zfs-replicate/default.nix [hunk 5]
```diff
@@ -1,19 +1,23 @@
-{ buildPythonApplication, click, fetchPypi, hypothesis, pytest
-, lib, stringcase
+{ buildPythonApplication
+, click
+, fetchPypi
+, hypothesis
```
Comment (pbsds) [inline pkgs/tools/backup/zfs-replicate/default.nix:4]:
```suggestion
, pytestCheckHook
, hypothesis
```

**Review by pbsds — CHANGES_REQUESTED** [review]

**Comment by alunduil** [comment]
@pbsds I think I got everything you mentioned, but let me know if there is anything else or if I misunderstood any of your suggestions.

File: pkgs/tools/backup/zfs-replicate/default.nix [hunk 6]
```diff
@@ -1,22 +1,28 @@
...
+, fetchPypi
+, pytestCheckHook
+, hypothesis
+, pytest
+, lib
+, stringcase
 }:
 
 buildPythonApplication rec {
   pname = "zfs-replicate";
```
Comment (pbsds) [inline pkgs/tools/backup/zfs-replicate/default.nix:12]: To match the fname on pypi:

```suggestion
  pname = "zfs_replicate";
```

File: pkgs/tools/backup/zfs-replicate/default.nix [hunk 7]
```diff
@@ -1,22 +1,28 @@
...
-  version = "1.2.3";
+  version = "3.1.0";
```
Comment (pbsds) [inline pkgs/tools/backup/zfs-replicate/default.nix:13]:
```suggestion
  version = "3.1.4";
```

File: pkgs/tools/backup/zfs-replicate/default.nix [hunk 8]
```diff
@@ -1,22 +1,28 @@
...
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "b2cb9d4670a6e12d14a446c10d857862e91af6e4526f607e08b41bde89953bb8";
+    hash = "sha256-24816c3e92ea9abc03e1ce281b5fcce4dfb5534092096e3e2c57a0d585ec4d67";
```
Comment (pbsds) [inline pkgs/tools/backup/zfs-replicate/default.nix:17]: ofborg eval error:
```
       error: invalid SRI hash '24816c3e92ea9abc03e1ce281b5fcce4dfb5534092096e3e2c57a0d585ec4d67'
```
---
```suggestion
    hash = "sha256-mRINo20/uFlWtP5W7w+D2E9o89hlAsqZmBjuv0qWP9k=";
```

**Review by pbsds — CHANGES_REQUESTED** [review]

**Comment by pbsds** [comment]
```
zfs_replicate> Executing setuptoolsBuildPhase
zfs_replicate> Traceback (most recent call last):
zfs_replicate>   File "/build/zfs_replicate-3.1.4/nix_run_setup", line 8, in <module>
zfs_replicate>     exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
zfs_replicate>   File "/nix/store/pzf6dnxg8gf04xazzjdwarm7s03cbrgz-python3-3.10.12/lib/python3.10/tokenize.py", line 394, in open
zfs_replicate>     buffer = _builtin_open(filename, 'rb')
zfs_replicate> FileNotFoundError: [Errno 2] No such file or directory: 'setup.py'
```

File: pkgs/tools/backup/zfs-replicate/default.nix [hunk 9]
```diff
@@ -1,22 +1,28 @@
...
+, pytest
+, lib
+, stringcase
 }:
 
 buildPythonApplication rec {
-  pname = "zfs-replicate";
-  version = "1.2.3";
+  pname = "zfs_replicate";
+  version = "3.1.4";
```
Comment (pbsds) [inline pkgs/tools/backup/zfs-replicate/default.nix:14]:
```suggestion
  pyproject = true;
```

**Comment by pbsds** [comment]
```
zfs_replicate> Creating a wheel...
zfs_replicate> * Getting build dependencies for wheel...
zfs_replicate> 
zfs_replicate> Traceback (most recent call last):
zfs_replicate>   File "/nix/store/xzdyc2wd3nrdn28iyr1q5maqfgcr56gm-python3.10-pyproject-hooks-1.0.0/lib/python3.10/site-packages/pyproject_hooks/_impl.py", line 321, in _call_hook
zfs_replicate>     raise BackendUnavailable(data.get('traceback', ''))
zfs_replicate> pyproject_hooks._impl.BackendUnavailable: Traceback (most recent call last):
zfs_replicate>   File "/nix/store/xzdyc2wd3nrdn28iyr1q5maqfgcr56gm-python3.10-pyproject-hooks-1.0.0/lib/python3.10/site-packages/pyproject_hooks/_in_process/_in_process.py", line 77, in _build_backend
zfs_replicate>     obj = import_module(mod_path)
zfs_replicate>   File "/nix/store/pzf6dnxg8gf04xazzjdwarm7s03cbrgz-python3-3.10.12/lib/python3.10/importlib/__init__.py", line 126, in import_module
zfs_replicate>     return _bootstrap._gcd_import(name[level:], package, level)
zfs_replicate>   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
zfs_replicate>   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
zfs_replicate>   File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
zfs_replicate>   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
zfs_replicate>   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
zfs_replicate>   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
zfs_replicate>   File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
zfs_replicate>   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
zfs_replicate>   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
zfs_replicate>   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
zfs_replicate>   File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
zfs_replicate>   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
zfs_replicate>   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
zfs_replicate>   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
zfs_replicate>   File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
zfs_replicate> ModuleNotFoundError: No module named 'poetry'
zfs_replicate> 
zfs_replicate> ERROR Backend 'poetry.core.masonry.api' is not available.
```

Please try to build it. The command is `nix-build . -A zfs-replicate`

File: pkgs/tools/backup/zfs-replicate/default.nix [hunk 10]
```diff
@@ -1,22 +1,29 @@
...
     inherit pname version;
-    sha256 = "b2cb9d4670a6e12d14a446c10d857862e91af6e4526f607e08b41bde89953bb8";
+    hash = "sha256-mRINo20/uFlWtP5W7w+D2E9o89hlAsqZmBjuv0qWP9k=";
   };
 
   postPatch = ''
-    sed -i setup.cfg \
-      -e '/--cov.*/d'
+    sed -i pyproject.toml -e '/--cov[^"]*/d'
   '';
```
Comment (pbsds) [inline pkgs/tools/backup/zfs-replicate/default.nix:24]:
```suggestion

  nativeBuildInputs = [
    poetry-core
  ];

```

Maybe `poetry` is needed

**Comment by alunduil** [comment]
> Please try to build it. The command is `nix-build . -A zfs-replicate`

Thanks. I had tried to remember this command but couldn't recall or find the exact one. I'll get this cleaned up with this verification.

**Comment by alunduil** [comment]
@pbsds, it looks like everything is good this time, but let me know if I need to include anything else.  Thanks for all of the support in cleaning this up and updating it.

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 250845` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>zfs-replicate</li>
    <li>zfs-replicate.dist</li>
  </ul>
</details>

---

- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] all depending packages build
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

**Review by pbsds — APPROVED** [review]

**Review by pbsds — CHANGES_REQUESTED** [review]
Could you reword the commit message to `zfs-replicate: 1.2.3 -> 3.1.4`

**Comment by alunduil** [comment]
@pbsds done.  Thanks!

**Review by pbsds — APPROVED** [review]

**Pull request merged**